### PR TITLE
Feature/id 370/lazy loading natural disasters

### PR DESCRIPTION
**Pull request by JCarlosHidalgoSosa**
implementation to UI and implementation about show incidents located in database.

### CODE REVIEW

File: app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import com.isc.hermes.utils.MapClickEventsManager;
+import com.isc.hermes.view.MapPolygonStyle;
+import com.mapbox.geojson.Point;
+
+import java.util.List;
+
+import android.os.Handler;
+import android.os.Looper;
+
+public class PolygonVisualizationController {
```
Comment (luiggy-mamani-jalafund) [inline app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java:15]: Add documentation for the classes and methods

File: app/src/main/java/com/isc/hermes/MainActivity.java [hunk 2]
```diff
@@ -100,6 +104,7 @@ protected void onCreate(Bundle savedInstanceState) {
         addIncidentGeneratorButton();
         MarkerManager.getInstance(this).removeSavedMarker();
         launcher = createActivityResult();
+        initShowIncidentsController();
         testPolyline(); // this is a test method that will be removed once the functionality has been verified.
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/MainActivity.java:108]: If the comment is important for a TODO and if not delete it

File: app/src/main/java/com/isc/hermes/ViewingPetitions.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package com.isc.hermes;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.isc.hermes.utils.MapClickEventsManager;
+import com.mapbox.mapboxsdk.maps.Style;
+
+public class ViewingPetitions extends AppCompatActivity {
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/ViewingPetitions.java]: class needs javadocs

File: app/src/main/java/com/isc/hermes/ViewingPetitions.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+    public void setMapStyle(){
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/ViewingPetitions.java]: method needs javadocs

File: app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java (hunk 1 above)
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java:15]: class needs javadocs

File: app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    private static PolygonVisualizationController instance;
+
+    public void displayPointsPolygonOnMap(List<List<Point>> allPolygonsPoints,String hexCodeColor)  {
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java:24]: method needs javadocs

File: app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        Handler handler = new Handler(Looper.getMainLooper());
+        handler.post(new Runnable() {
+            @Override
+            public void run() {
+                new MapPolygonStyle(MapClickEventsManager.getInstance().getMapboxMap(), allPolygonsPoints,hexCodeColor);
+            }
+        });
+    }
+
+    public static PolygonVisualizationController getInstance(){
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java:38]: method needs javadocs

File: app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+import android.view.View;
+import android.widget.Button;
+import android.widget.ImageButton;
+import android.widget.LinearLayout;
+import androidx.appcompat.app.AppCompatActivity;
+import com.isc.hermes.R;
+import com.isc.hermes.model.incidentsRequesting.NaturalDisasterRequesting;
+
+
+public class ViewIncidentsController{
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java:14]: class needs javadocs.

File: app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        this.viewIncidentsFormButton = activity.findViewById(R.id.viewIncidentsButton);
+        this.viewTrafficButton = activity.findViewById(R.id.viewTrafficButton);
+        this.viewSocialEventButton = activity.findViewById(R.id.viewSocialIncidentsButton);
+        this.viewNaturalDisasterButton = activity.findViewById(R.id.viewNaturalDisasterButton);
+        initEyeButtonFunctionality();
+        initViewDifferentIncidentsTypeButton();
+    }
+
+
+    private void initEyeButtonFunctionality() {
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java:38]: method needs javadocs.

File: app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+        viewIncidentsFormButton.setOnClickListener(v -> {
+            if (!buttonMarked) incidentsVisualizationForm.setVisibility(View.VISIBLE);
+            else incidentsVisualizationForm.setVisibility(View.GONE);
+            buttonMarked = !buttonMarked;
+        });
+    }
+
+    private void initViewDifferentIncidentsTypeButton() {
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java:49]: method needs javadocs.

File: app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+            disappearForm();
+        });
+
+        viewNaturalDisasterButton.setOnClickListener(v -> {
+            PolygonVisualizationController.getInstance().displayPointsPolygonOnMap(requesting.getAllPolygonPoints(),"#Ff0000");
+            disappearForm();
+        });
+    }
+
+    private void disappearForm(){
```
Comment (VillalobosGarciaGaby) [inline app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java:68]: method needs javadocs.

File: app/src/main/java/com/isc/hermes/MainActivity.java (hunk 2 above)
Comment (JCarlosHidalgoSosa) [reply app/src/main/java/com/isc/hermes/MainActivity.java:108]: that's a work implemented from another teammate, I don't know the consequences about detete it.

Comment (VillalobosGarciaGaby) [reply app/src/main/java/com/isc/hermes/MainActivity.java:108]: then just put it as TODO

File: app/src/main/java/com/isc/hermes/ViewingPetitions.java (hunk 3 above)
Comment (JCarlosHidalgoSosa) [reply app/src/main/java/com/isc/hermes/ViewingPetitions.java]: Already writed.

File: app/src/main/java/com/isc/hermes/ViewingPetitions.java (hunk 4 above)
Comment (JCarlosHidalgoSosa) [reply app/src/main/java/com/isc/hermes/ViewingPetitions.java]: Already writed.

File: app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java (hunk 1 above)
Comment (JCarlosHidalgoSosa) [reply app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java:15]: Already writed.

File: app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java (hunk 5 above)
Comment (JCarlosHidalgoSosa) [reply app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java:24]: Already writed.

File: app/src/main/java/com/isc/hermes/ViewingPetitions.java (hunk 4 above)
Comment (JCarlosHidalgoSosa) [reply app/src/main/java/com/isc/hermes/ViewingPetitions.java]: unnecessary class, sorry, already deleted.

File: app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java (hunk 6 above)
Comment (JCarlosHidalgoSosa) [reply app/src/main/java/com/isc/hermes/controller/PolygonVisualizationController.java:38]: Already writed.

File: app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java (hunk 7 above)
Comment (JCarlosHidalgoSosa) [reply app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java:14]: Already writed.

File: app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java (hunk 8 above)
Comment (JCarlosHidalgoSosa) [reply app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java:38]: Already writed.

File: app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java (hunk 9 above)
Comment (JCarlosHidalgoSosa) [reply app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java:49]: Already writed.

File: app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java (hunk 10 above)
Comment (JCarlosHidalgoSosa) [reply app/src/main/java/com/isc/hermes/controller/ViewIncidentsController.java:68]: Already writed.

**Review by luiggy-mamani-jalafund — APPROVED** [review]

**Review by daniel-espinoza-79 — APPROVED** [review]
good job

**Review by VillalobosGarciaGaby — APPROVED** [review]

**Comment by VillalobosGarciaGaby** [comment]
good job

**Pull request merged**